frontend/src/components/Pages/MissionHistoryPage/MissionHistoryView.tsx

```diff
@@ -33,31 +63,59 @@ export function MissionHistoryView({ refreshInterval }: RefreshProps) {
     const [isLoading, setIsLoading] = useState<boolean>(true)
     const [isResettingPage, setIsResettingPage] = useState<boolean>(false)
 
-    const {
-        page,
-        switchPage,
-        missionName,
-        statuses,
-        robotName,
-        tagId,
-        inspectionTypes,
-        minStartTime,
-        maxStartTime,
-        minEndTime,
-        maxEndTime,
-    } = useMissionFilterContext()
+    const { page, switchPage, filterState, filterIsSet, filterFunctions, filterError, clearFilterError } =
+        useMissionFilterContext()
+
+    const FilterErrorDialog = () => {
+        return (
+            <>
+                <Dialog open={filterError !== ''} isDismissable onClose={() => clearFilterError()}>
+                    <Dialog.Header>
+                        <Dialog.Title>Dismissable dialog</Dialog.Title>
+                    </Dialog.Header>
+                    <Dialog.CustomContent>
+                        <Typography variant="body_short">Closes dialog on click outside and escape key.</Typography>
+                    </Dialog.CustomContent>
+                    <Dialog.Actions>
+                        <Button onClick={() => clearFilterError()}>OK</Button>
+                    </Dialog.Actions>
+                </Dialog>
+            </>
+        )
+    }
+
+    const checkBoxBackgroundColour = tokens.colors.ui.background__info.hex
+    const checkBoxBorderColour = tokens.colors.interactive.primary__resting.hex
+
+    const toDisplayValue = (filterName: string, value: string | number | MissionStatus[] | InspectionType[]) => {
+        if (typeof value === 'string') {
+            return value
+        } else if (typeof value === 'number') {
+            // We currently assume these are dates. We may want
+            // to explicitly use the Date type in the filter context instead
+            return filterFunctions.dateTimeIntToPrettyString(value)
+        } else if (Array.isArray(value)) {
+            let valueArray = value as any[]
+            if (valueArray.length === 0) {
+                console.error('Unexpected empty array detected')
+            }
+            return valueArray.map((val) => (
+                <Chip
```

Comment:
I think I liked it better without chip inside the chip

Reply:
I'm not sure if I understand, since before adding the chip inside the chip we couldn't delete specific values in arrays, but only the whole array?

Reply:
I have implemented your CSS suggestions so I hope you're OK with marking this as resolved